src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ExplicitBodyParam_ComplexReturn_Snapshot.generated.txt

```diff
@@ -299,6 +299,13 @@ namespace Microsoft.AspNetCore.Http.Generated
             }
             return (false, default);
         }
+
+        private static StringValues ResolveFromRouteOrQuery(HttpContext httpContext, string parameterName, IEnumerable<string>? routeParameterNames)
```

Comment:
Ah, interesting point. It doesn't get used in this particular test scenario. We can probably do something that only emits the `ResolveFromRouteOrQuery` method if we ended up doing route-vs-query inference in the generated code.

We would have to do some bookkeeping to track when that has happened.

Alternatively, it might be easier to just inline this resolution check.